web/src/components/overview/StorageSection.jsx

```diff
@@ -29,19 +29,39 @@ import { deviceLabel } from "~/components/storage/utils";
 import { Em, ProgressText, Section } from "~/components/core";
 import { _ } from "~/i18n";
 
+/**
+ * Text explaining the storage proposal
+ *
+ * FIXME: this needs to be basically rewritten. See
+ * https://github.com/openSUSE/agama/discussions/778#discussioncomment-7715244
+ */
 const ProposalSummary = ({ proposal }) => {
   const { availableDevices = [], result = {} } = proposal;
 
   const bootDevice = result.settings?.bootDevice;
   if (!bootDevice) return <Text>{_("No device selected yet")}</Text>;
 
   const device = availableDevices.find(d => d.name === bootDevice);
-
   const label = device ? deviceLabel(device) : bootDevice;
 
-  // TRANSLATORS: %s will be replaced by the device name and its size,
-  // example: "/dev/sda, 20 GiB"
-  const [msg1, msg2] = _("Install using device %s and deleting all its content").split("%s");
+  const fullMsg = (policy) => {
+    switch (policy) {
+      case "resize":
+        // TRANSLATORS: %s will be replaced by the device name and its size,
+        // example: "/dev/sda, 20 GiB"
+        return _("Install using device %s shrinking existing partitions as needed");
+      case "keep":
+        // TRANSLATORS: %s will be replaced by the device name and its size,
+        // example: "/dev/sda, 20 GiB"
+        return _("Install using device %s without modifying existing partitions");
+    }
+    // TRANSLATORS: %s will be replaced by the device name and its size,
+    // example: "/dev/sda, 20 GiB"
+    return _("Install using device %s and deleting all its content");
```

Comment:
to be honest I am not fan of those implivit fallthrough for enum values. I would add case also for `"delete"` so if we support custom in future it can properly write to console something like "unknown value" and you future Ivan will thank you.

Reply:
Sure. But take into account the full `ProposalSummary` component is meant to be rewritten in the short term.